test/metabase/related_test.clj

```diff
@@ -0,0 +1,75 @@
+(ns metabase.related-test
+  (:require [expectations :refer :all]
+            [metabase.api.common :as api]
+            [metabase.models
+             [card :refer [Card]]
+             [database :refer [Database]]
+             [table :refer [Table]]]
+            [metabase.related :refer :all :as r]
+            [metabase.test.util :as tu]
+            [metabase.test.data.users :as users]
+            [toucan.util.test :as tt]))
+
+(expect
+  #{[:field-id 1] [:metric 1] ["FIELD-ID" 2] ["segment" 1]}
+  (#'r/collect-context-bearing-forms [[:> [:field-id 1] 3]
+                                      ["and" [:= ["FIELD-ID" 2] 2]
+                                       ["segment" 1]]
+                                      [:metric 1]]))
+
+(expect
+  [0.5
+   0.0
+   1.0]
+  (tt/with-temp* [Card [{card-id-1 :id}
+                        {:name          (tu/random-name)
```

Comment:
You don't need to add `{:name (tu/random-name)` here. It's part of the defaults for temp Cards:

```
(u/strict-extend (class Card)
  test/WithTempDefaults
  {:with-temp-defaults (fn [_] {:creator_id             (rasta-id)
                                :dataset_query          {}
                                :display                :table
                                :name                   (random-name)
                                :visualization_settings {}})})
```

Just make sure that the `metabase.test.util` namespace is loaded, otherwise that won't work. So make sureto `:require` this even if you're not using any of its functions, or `lein test metabase.related-test` (for example) won't work. 

Actually `metabase.test-setup` is automatically loaded whenever running tests, so if you wanted to add a `:require` for `metabase.test.util` you wouldn't need to remember to add the requires in every test namespace that implicitly uses it.

Reply:
Fixed. tnx